File: src/main/java/waterpower/common/item/crafting/MaterialForms.java
/**
 * Copyright (c) Huang Yuhui, 2014
 * 
 * "WaterCraft" is distributed under the terms of the Minecraft Mod Public
 * License 1.0, or MMPL. Please check the contents of the license located in
 * http://www.mod-buildcraft.com/MMPL-1.0.txt
 */

package waterpower.common.item.crafting;

import net.minecraft.client.resources.I18n;

public enum MaterialForms {
    plate, dust, dustTiny, dustSmall, plateDense, ingot, stick, screw, gear, nugget, ring;

    public String getShowedName() {
        String format = "cptwtrml.forms." + name();
        String s = I18n.format(format);
        return s;
    }
}

File: src/main/kotlin/waterpower/common/item/MaterialForms.kt
/**
 * Copyright (c) Huang Yuhui, 2017
 *
 * "WaterPower" is distributed under the terms of the Minecraft Mod Public
 * License 1.0, or MMPL. Please check the contents of the license located in
 * http://www.mod-buildcraft.com/MMPL-1.0.txt
 */
package waterpower.common.item

import net.minecraft.util.IStringSerializable

enum class MaterialForms : IStringSerializable {
    dustTiny, dustSmall, dust, plate, plateDense, nugget, ingot, stick, screw, gear, ring;

    override fun getName() = name

    fun getLocalizedName() = waterpower.client.i18n("waterpower.forms.${getName()}")
}